draft-ietf-ppm-dap.md

```diff
@@ -1338,11 +1362,11 @@ following generic checks.
       store the set of IDs of processed reports. However, implementations may
       find it helpful to track additional information, like the timestamp, so
       that the storage used for anti-replay can be sharded efficiently.
+    * This also covers report re-use for the Prio3 VDAF because its aggregation
+      parameter is nil.
 
-1. Check if the report has never been aggregated but is contained by a batch
-   that has been collected. If this check fails, the input share MUST be marked
-   as invalid with the error `batch_collected`. This prevents additional reports
-   from being aggregated after its batch has already been collected.
```

Comment:
Since the previous n which have been aggregated cannot be used anymore this is already covered. The only way to aggregate again is if there are another $threshold reports for the same time interval. In that case it's fine.

Maybe the phrasing is indeed the problem. Afaiu "is in a batch which has been collected" and "has never been aggregated" are contradictory. I assumed it meant "batch definition" i.e. "time interval".

Reply:
> Since the previous n which have been aggregated cannot be used anymore this is already covered. The only way to aggregate again is if there are another $threshold reports for the same time interval. In that case it's fine.

Hmm, I think I'm missing something important here. Can you clarify? Is there language elsewhere that covers this?

So the requirement is: "once a batch has been defined, do not add any more reports to that batch".

Reply:
Okay, so how about
```
   A report must never be used in more than one batch. If a report with the same report ID is contained in a batch    for which the `AggregateShare` has been sent the input share MUST be marked as invalid with the error `batch_collected`.
```
This makes the requirement that Chris stated explicit, and removes the somewhat contradictory statement about a batch which has been collected but contains a report which has not been aggregated.

Reply:
I don't think that phrasing is right, because to me it doesn't capture that you can't add a report to a batch once it has been collected. How about this:

```
1. Some VDAFs allow a batch to be collected multiple times with different aggregation parameters, but the aggregators must check that the same set of reports is being aggregated each time. A report may not be aggregated into a collection unless it was included in all previous collections for the batch. If this check fails, the input share MUST be marked as invalid with the error `batch_collected`. This prevents collectors from learning anything about small numbers of reports that are uploaded between two collections of a batch.
```

Reply:
I was trying to clarify what "once it has been collected" means with "for which the `AggregateShare` has been sent". Since the helper cannot tell if collection has actually finished (and if there is pro-active aggregation maybe even if it has started) it can only check if it has sent the `AggregateShare` to the Leader.